ra/ra.go

```diff
@@ -788,9 +788,13 @@ func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, authzs []*c
 				return
 			}
 
+			// TODO(jsing): This needs to be generated based on the entry
+			// point to the service...
+			accountURI := fmt.Sprintf("http://boulder:4000/acme/reg/%d", authz.RegistrationID)
```

Comment:
Looking for some guidance here - I'm not sure what the best option is for regenerating the full account URI, since it depends on the service domain name and the ACME version... there did not appear to be an obvious way to determine/reconstruct this. Same goes for the `va/va.go` change.

Reply:
I think we need to add a config option the the VA with a list of account URI prefixes that are accepted. This will cover the fact that we consider v1 and v2 account URIs synonymous.

I think if we pass the account ID to the VA instead of the account URI, we only need to add a config option to VA, not RA. Do you think that makes sense?

Reply:
Sure - that works. In that case we just pass the account ID (aka registration ID) through the gRPC call and reconstruct the actual account URIs at check time based off the configured prefixes. I've reworked the code accordingly.